specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/dedicatedhsm.json

```diff
@@ -0,0 +1,762 @@
+          "Create a new or update an existing payment HSM with management profile": {
+            "$ref": "./examples/PaymentHsm_CreateOrUpdate_WithManagementProfile.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "DedicatedHsms"
+        ],
+        "operationId": "DedicatedHsm_Update",
+        "x-ms-long-running-operation": true,
+        "description": "Update a dedicated HSM in the specified subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DedicatedHsmName"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmPatchParameters"
+            },
+            "description": "Parameters to patch the dedicated HSM"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Patched dedicated HSM",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update an existing dedicated HSM": {
+            "$ref": "./examples/DedicatedHsm_Update.json"
+          },
+          "Update an existing payment HSM": {
+            "$ref": "./examples/PaymentHsm_Update.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "DedicatedHsms"
+        ],
+        "operationId": "DedicatedHsm_Delete",
+        "x-ms-long-running-operation": true,
+        "description": "Deletes the specified Azure Dedicated HSM.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DedicatedHsmName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK Response."
+          },
+          "202": {
```

Comment:
If delete is async, 200 response is not used.

Reply:
Yes, only 202 and 204 should be needed.

Reply:
Thanks! Removed 200 code.